<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [3]:
# Membuat dataframes dari dictionary
import pandas as pd

hewan = {
    'Nama':['Ayam','Singa','Sapi','Elang'],
    'Kaki':[2,4,4,2],
    'Jenis':['Omnivora','Karnivora','Herbivora','Karnivora']
}

df = pd.DataFrame(hewan)
df

Nama  Kaki      Jenis
0   Ayam     2   Omnivora
1  Singa     4  Karnivora
2   Sapi     4  Herbivora
3  Elang     2  Karnivora

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [8]:
# Membaca data dan memasukannya kedalam suatu variabel
df_ = pd.read_csv('/content/drive/My Drive/Bootcamp Data Science Python/Pekan 2/1. Pandas Introduction/diamonds-1.csv')

df_



Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 

**Perbedaan Series dengan list dan dictionary:**

Series mempunyai index yang dapat difungsikan sebagai key, sedangkan pada list hanya memiliki index dan pada dictionary hanya memiliki key. Serta index pada series dapat disesuaikan dengan keinginan/kebutuhan kita.

**Perbedaan dataframe dan series:**
*   Series adalah objek 1 dimensi
*   DataFrame adalah suatu objek 2 dimensi, mirip seperti tabel

In [9]:
# code here
buah = pd.Series(['mangga','apel','kiwi','jeruk'])
print(buah)

0    mangga
1      apel
2      kiwi
3     jeruk
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [17]:
# code here
print('Baris pertama dari kolom price: ',df_.iloc[0]['price'])
print('Baris terakhir dari kolom color: ',df_.iloc[-1]['color'])

Baris pertama dari kolom price:  326
Baris terakhir dari kolom color:  D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 

Deskriptif statistik digunakan untuk menggambarkan/summary data. Seperti : nilai mean, median, modus, variance, standar deviasi, quartile, range dan sejenisnya. Descriptive statistics berlaku ketika pengumpulan seluruh sample data (populasi) dapat dilakukan.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [21]:
# code here

print(df_.describe())
print('\n',df_['price'].mean())
print('\n',df_['depth'].std())
print('\n',df_['carat'].max())

         Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000     

Jawab:

Rata-rata dar kolom price: 3932.799721913237

Standar deviasi dari kolom depth:  1.4326213188337733

Nilai maksimum dari kolom carat:  5.01